src/routes/estimations/estimations.service.ts

```diff
@@ -34,37 +34,14 @@ export class EstimationsService {
       chainId: args.chainId,
       address: args.address,
     });
-    const recommendedNonce = await this.getEstimationRecommendedNonce({
+    const recommendedNonce = await this.safeRepository.getRecommendedNonce({
```

Comment:
`EstimationsService.getEstimation` execution would call `SafeRepository.getSafe` twice if I'm not wrong (in line 33, and also inside `SafeRepository.getRecommendedNonce`). I know source data should be cached on the second call, but even considering this, I think it's better to avoid the duplicated call.

It could make sense to have `safe` as an optional argument for `SafeRepository.getRecommendedNonce`, so the caller could provide the `Safe` or not, and  `SafeRepository.getRecommendedNonce` would call `SafeRepository.getSafe` only when needed. What do you think?

Reply:
Yes the reason why I decided to use the Safe address and not the Safe instance is because from an interface perspective (specially one exposed to the routes) we have consistency in the functionality that we provide. See https://github.com/safe-global/safe-client-gateway/blob/recommended-nonce-endpoint/src/domain/safe/safe.repository.interface.ts

If `safe` is then an optional argument at the interface level it can be more confusing in my opinion when developing new features – calling this function would probably raise the question if the Safe info should be passed to it or not which deviates from the other functions.

A better alternative in my opinion would be to return a tuple of `[recommendedNonce, SafeInfo]` (and the Estimations Controller would use both). I didn't do this because I would say that there were not so many benefits to it – that use case is specific to the Estimations Controller and given that we are in the `ISafeRepository` we should assume that it'd have access to the respective `SafeInfo` domain entity.

Reply:
Thanks for the detailed explanations. I think it makes sense from an architecture perspective to favor clarity over performance, and I get your point about having access to the `SafeInfo` entity. I also thought about changing the return type to a tuple including the `SafeInfo`, but I think there is no perfect solution as you said, so I think reading from the cache on the second call is acceptable for the sake of simplicity on the API contract 👍

Reply:
Performance was also taken into consideration. We do have one more cache-read when calling the estimations endpoint but I think that is acceptable given the scope.